src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java

```diff
@@ -432,4 +432,153 @@ private String validateSubSectorCreateReq(Map<String, Object> sectorRequest) {
+			missingParams.add(refType);
+		}
+		if (CollectionUtils.isNotEmpty(missingParams)) {
+			errMsg = "Request is missing one or more parameters. Missing params : " + missingParams.toString();
+		}
+		return errMsg;
+	}
+
+	public Map fetchDesignationDetails(String Id) {
+		Map<String,Object> result = null;
+		try {
+			StringBuilder strUrl = new StringBuilder(extServerProperties.getPoresServiceHost());
+			strUrl.append(extServerProperties.getDesignationApiUrl()).append("/").append(Id);
+			log.info("printing api: "+strUrl.toString());
+			Map<String, String> map = new HashMap<String, String>();
+			map.put(Constants.CONTENT_TYPE_KEY,Constants.APPLICATION_JSON);
+			Map<String, Object> apiResponse = (Map<String, Object>) outboundRequestHandlerServiceImpl.fetchUsingGetWithHeaders(strUrl.toString(),map);
+			if (null != apiResponse) {
+				if (Constants.OK.equalsIgnoreCase((String) apiResponse.get(Constants.RESPONSE_CODE))) {
+					Map<String, Object> resultMap = (Map<String, Object>) apiResponse.get(Constants.RESULT);
+					result = (Map<String, Object>) resultMap.get(Constants.RESULT);
+					log.info("fetchDesignationDetails reulst: "+result);
+					return result;
+				} else {
+					log.info("Failed to fetch Designation Details: ");
+				}
+			} else {
+				log.error("Failed to read the des details for Id : " + Id);
+			}
+		} catch (Exception e) {
+			log.error("Failed to read Designation with Id: " + Id, e);
+		}
+		return result;
+	}
+
+	public SBApiResponse updateDesignation(String id, String designation, List<String> refNodes) {
+		SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_DESIGNATION_UPDATE);
+		try {
+			String apiUrl = extServerProperties.getPoresServiceHost() + extServerProperties.getDesignationUpdateUrl();
+
+			Map map = createRequestBody(id, designation, refNodes);
+
+			Map<String,Object> result = outboundRequestHandlerServiceImpl.fetchResultUsingPut(apiUrl,map,null);
+			if (result != null
+					&& Constants.OK.equalsIgnoreCase((String) result.get(Constants.RESPONSE_CODE))) {
+				Map<String, Object> resultMap = (Map<String, Object>) result.get(Constants.RESULT);
+				log.info("update Designation with identifiers: "+resultMap);
+				response.setResponseCode(HttpStatus.OK);
+			} else {
+				response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+				response.getParams().setErrmsg("Failed to update the designation for id: " + id);
+			}
+		} catch (Exception e) {
+			log.error("Failed to update designation with id: " + id, e);
+			response.getParams().setErrmsg("Failed to update designation: " + e.getMessage());
+			response.getParams().setStatus(Constants.FAILED);
+			response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		return response;
+	}
+
+	private Map<String, Object> createRequestBody(String id, String designation, List<String> refNodes) {
+		try {
+			Map<String, Object> reqBodyMap = new HashMap<>();
+			reqBodyMap.put(Constants.ID, id);
```

Comment:
why REF_ID and REF_TYPE are not used here ?
If those are not required then why those are mandatory?

Reply:
id is ref_id only here. we are passing from caller function.